cmeutils/sampling.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from pymbar import timeseries
+
+def equil_sample(
+        data, threshold_fraction=0.0, threshold_neff=1, conservative=True
+    ):
+    """Returns a statistically independent subset of an array of data.
+
+    Parameters
+    ----------
+    data : numpy.typing.Arraylike
+        1-D time dependent data to check for equilibration.
+    threshold_fraction : float, optional, default=0.8
+        Fraction of data expected to be equilibrated.
+    threshold_neff : int, optional, default=100
+        Minimum amount of effectively correlated samples to consider a_t
+        'equilibrated'.
+    conservative : bool, default=True
+        if set to True, uniformly-spaced indices are chosen with interval
+        ceil(g), where g is the statistical inefficiency.  
+        Otherwise, indices are chosen non-uniformly with interval of
+        approximately g in order to end up with approximately T/g total indices
+
+    Returns
+    -------
+    Tuple containing
+
+    """
+    is_equil, prod_start, ineff, Neff = is_equilibrated(
+            data, threshold_fraction, threshold_neff
+            )
+
+    if is_equil is True:
+        uncorr_indices = timeseries.subsampleCorrelatedData(
+                data[prod_start:], g=ineff, conservative=conservative
+            )
+        uncorr_sample = data[prod_start:][uncorr_indices]
+        return(uncorr_sample, uncorr_indices, prod_start, Neff)
+
+    elif is_equil is False:
+        raise ValueError(
+            "Property does not have requisite threshold of production data "
+            "expected. More production data is needed, or the threshold needs "
+            "to be lowered. See is_equilibrated for more information."
+        )
+
+def is_equilibrated(data, threshold_fraction=0.80, threshold_neff=100, nskip=1):
```

Comment:
this threshold might be a bit high. In reproducibility study we found we needed to lower it from 0.75 to get reasonable run times.

Reply:
Maybe set 0.50 as a default?